casper/src/main/resources/Pos.rhox

```diff
@@ -111,616 +114,689 @@ in {
+                                        // Moves slashed validator to withdrawers map with no quarantine period.
+                                        stateUpdateCh!({
+                                          "allBonds"        : state.get("allBonds").set(slashedValidator, 0),
+                                          "activeValidators": state.get("activeValidators").delete(slashedValidator),
+                                          "committedRewards" : state.get("committedRewards").delete(slashedValidator),
+                                          "withdrawers"     : state.get("withdrawers"),
+                                          "pendingWithdrawers": state.get("pendingWithdrawers"),
+                                          "randomImages"    : state.get("randomImages"),
+                                          "randomNumbers"   : state.get("randomNumbers")
+                                        },
+                                        (true, Nil))
                                       }
                                     }
-                                  } |
-                                  // Checks withdrawer's quarantine period against current block number,
-                                  // then computes their payment, removes them from the bonds and withdrawers maps, and
-                                  // updates the committed rewards map.
-                                  contract removeQuarantinedWithdrawers(@currentBlockNumber, @state, returnCh) = {
-                                    new quarantinedValidatorsCh, validatorsToWithdrawListCh,
-                                        isQuarantineFinished, notWithdrawn, payWithdraw,
-                                        newBondsListCh, newWithdrawersListCh,
-                                        computeRemove, payRet
+                                  }
+                                }
+                              }
+                            }
+                          } else {
+                            returnCh!((false, "Invalid system auth token"))
+                          }
+                        }
+                      }
+                    } |
+                    // Private method which signals the end of block processing.
+                    contract PoS(@"closeBlock", @sysAuthToken, ackCh) = {
+                      new isValidTokenCh in {
+                        sysAuthTokenOps!("check", sysAuthToken, *isValidTokenCh) |
+                        for (@isValid <- isValidTokenCh) {
+                          if (isValid) {
+                            new blockDataCh, getBlockData(`rho:block:data`) in {
+                              getBlockData!(*blockDataCh) |
+                              for (@epochBlockNumber <<- epochBlockNumberCh){
+                                for (@blockNumber, _ <- blockDataCh) {
+                                  if (epochBlockNumber != blockNumber) {
```

Comment:
In manual bonding procedure, which we want to use initially, `epochLength` should not be used and `epochBlockNumber` `Nil` value should be used to represent infinite value meaning no epoch change will happen.

When new validator requests are waiting or when withdrawal requests are initiated, Coop will decide on which block number epoch should happen and will set `epochBlockNumber`. After epoch change is done `epochBlockNumber` should again be reset to `Nil`.

Reply:
Got it. So we initially set it to `NIl` and as long as `epochBlockNumber` is either `Nil` or an int greater than the current block number, no epoch change happens. We can also make it so that if `epochBlockNumber` is an int less than the current block  number, an epoch change will occur if `(blockNumber - epochBlockNumber) % == 0`